src/v2/Apps/Artist/Components/ArtistConsignButton.tsx

```diff
@@ -15,14 +18,35 @@ import {
   color,
 } from "@artsy/palette"
 import { AnalyticsSchema, useTracking } from "v2/Artsy"
+import { useFragment } from "relay-hooks"
 
 export interface ArtistConsignButtonProps {
-  artist: ArtistConsignButton_artist
+  artist: ArtistConsignButton_artist$key
 }
 
-export const ArtistConsignButton: React.FC<ArtistConsignButtonProps> = ({
-  artist,
-}) => {
+export const ArtistConsignButton: React.FC<ArtistConsignButtonProps> = props => {
+  const artist = useFragment<ArtistConsignButton_artist$key>(
```

Comment:
thought: do you need the generic thing in there? I think you can skip it and ts will figure it out. I also like to have the fragment defined above the component, do it's not inside and taking all that space.

Reply:
I'll play with it a bit more -- one would think, since the second arg comes directly from the type 🤔

Reply:
Yep, no need 👍

Reply:
> I also like to have the fragment defined above the component, do it's not inside and taking all that space

My thinking there is that the component is slighly more portable with stuff defined in the function, but that patttern with queries is not too common so moved it out.